vnext/ReactUWP/Views/FlyoutViewManager.cpp

```diff
@@ -155,10 +156,48 @@ void FlyoutShadowNode::createView() {
   m_flyoutClosedRevoker =
       m_flyout.Closed(winrt::auto_revoke, [=](auto &&, auto &&) {
         auto instance = wkinstance.lock();
-        if (!m_updating && instance != nullptr)
+        if (!m_updating && instance != nullptr) {
+          if (m_targetElement != nullptr) {
+            // When the flyout closes, attempt to move focus to
+            // its anchor element to prevent cases where focus can land on
+            // an outer flyout content and therefore trigger a unexpected flyout
+            // dismissal
+            winrt::FocusManager::TryFocusAsync(
+                m_targetElement, winrt::FocusState::Programmatic);
+          }
+
           OnFlyoutClosed(*instance, m_tag, false);
+        }
       });
 
+  // Turn off AllowFocusOnInteraction so that focus cannot land in the top
+  // content element of the flyout. If focus lands in the flyout presenter, then
+  // on moving focus somewhere else, including the children on the flyout
+  // presenter, the flyout can dismiss.
+  //
+  // However, turning AllowFocusOnInteraction off at the root of the flyout and
+  // flyout presenter turns it off for all children of the flyout. In order to
+  // make sure that interactions with the content of the flyout still work as
+  // expected, AllowFocusOnInteraction is turned on for all children of the
+  // flyout presenter.
+  m_flyout.AllowFocusOnInteraction(false);
+
+  m_tokenContentPropertyChangeCallback =
+      m_flyout.RegisterPropertyChangedCallback(
+          winrt::Flyout::ContentProperty(),
+          [=](winrt::DependencyObject sender, winrt::DependencyProperty dp) {
+            if (auto flyout = sender.try_as<winrt::Flyout>()) {
+              if (auto content = flyout.Content()) {
+                auto children = content.GetChildrenInTabFocusOrder();
```

Comment:
Will setting of AllowFocusOnInteraction on the content of the flyout enable it for all the children? Or do we need to cherry pick some children using GetChildrenInTabFocusOrder and enable it on only those? (Not sure what the difference is...)

Reply:
Ah the content works actually. I was worried that setting AllowFocusOnInteraction on the Content directly would land us in the same problem with the flyout dismissing if you clicked on a blank part of the flyout, but it seems to work just fine. Let me update that.